pkgs/development/compilers/solc/default.nix

```diff
@@ -1,28 +1,30 @@
 { stdenv, fetchzip, fetchgit, boost, cmake }:
 
+let rev = "194ff033ae44944ac59aa7bd3da89ba94ec5893c"; in
+let sha256 = "0gkg3nay0625qmhxxxax1d1c4dl554ri3pkwd12qfg6g1w6j04w7"; in
+let version = "0.4.12"; in
+
+let jsoncppURL = https://github.com/open-source-parsers/jsoncpp/archive/1.7.7.tar.gz; in
 let jsoncpp = fetchzip {
-  url = https://github.com/open-source-parsers/jsoncpp/archive/1.7.7.tar.gz;
+  url = jsoncppURL;
   sha256 = "0jz93zv17ir7lbxb3dv8ph2n916rajs8i96immwx9vb45pqid3n0";
```

Comment:
Just a small stylistic comment - was there a reason to prefer this style over:

```
let rev = "194ff033ae44944ac59aa7bd3da89ba94ec5893c";
    sha256 = "0gkg3nay0625qmhxxxax1d1c4dl554ri3pkwd12qfg6g1w6j04w7";
    version = "0.4.12";
    jsoncppURL = https://github.com/open-source-parsers/jsoncpp/archive/1.7.7.tar.gz;
    ... rest of the bindings ...
in
```

which I find bit cleaner compared to nested `let` expressions.